File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.java
class Solution {
    public int[] solution(int n, int[] slicer, int[] num_list) {
        int[] answer = {};
        int a = slicer[0];
        int b = slicer[1];
        int c = slicer[2];
        int index = 0;
        
        switch(n){          
            case 1:
                answer = new int[b+1];
                for(int i = 0; i <= b; i++){
                    answer[i] = num_list[i];
                }
                break;
            case 2:
                answer = new int[num_list.length-a];
                for(int i = a; i < num_list.length; i++){
                    answer[index] = num_list[i];
                    index++;
                }
                break;
            case 3:
                answer = new int[b-a+1];
                for(int i = a; i <= b; i++){
                    answer[index] = num_list[i];
                    index++;
                }
                break;
            case 4:
                answer = new int[(b-a)/c+1];
                for(int i = a; i <= b; i+=c){
                    answer[index] = num_list[i];
                    index++;
                }
                break;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// slicer_len은 배열 slicer의 길이입니다.
// num_list_len은 배열 num_list의 길이입니다.
int* solution(int n, int slicer[], size_t slicer_len, int num_list[], size_t num_list_len) {
    int a, b, c;
    int len;
    int idx = 0;
    
    if(n == 1) {
        a = 0;
        b = slicer[1];
        c = -1;
        len = b + 1;
    }else if(n == 2) {
        a = slicer[0];
        b = num_list_len;
        c = -1;
        len = b - a;
    }else if(n == 3) {
        a = slicer[0];
        b = slicer[1];
        c = -1;
        len = b - a + 1;
    }else {
        a = slicer[0];
        b = slicer[1];
        c = slicer[2];
        len = (a + b) / c;
    }
    
    int* answer = (int*)malloc(sizeof(int) * len);
    
    for(; a <= b;){
        answer[idx++] = num_list[a];
        if(c > 0){
            a += c;
        }else {
            a++;
        }
    }
    
    return answer;
}